### PR TITLE
adding SonarCube to build

### PR DESCRIPTION
**Pull request by partnerinflight**
all good :)

### CODE REVIEW

**Comment by partnerinflight** [comment]
SonarQube analysis reported 23 issues
* ![BLOCKER][BLOCKER] 18 blocker
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 2 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] [build.py#L90](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/8a044343e0a9d5a4497e7bd0f6fd832f8b334a59/build.py#L90): Replace print statement by built-in function. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3APrintStatementUsage)
1. ![BLOCKER][BLOCKER] [build.py#L91](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/8a044343e0a9d5a4497e7bd0f6fd832f8b334a59/build.py#L91): Replace print statement by built-in function. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3APrintStatementUsage)
1. ![BLOCKER][BLOCKER] [build.py#L94](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/8a044343e0a9d5a4497e7bd0f6fd832f8b334a59/build.py#L94): Replace print statement by built-in function. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3APrintStatementUsage)
1. ![BLOCKER][BLOCKER] [build.py#L97](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/8a044343e0a9d5a4497e7bd0f6fd832f8b334a59/build.py#L97): Replace print statement by built-in function. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3APrintStatementUsage)
1. ![BLOCKER][BLOCKER] [build.py#L99](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/8a044343e0a9d5a4497e7bd0f6fd832f8b334a59/build.py#L99): Replace print statement by built-in function. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3APrintStatementUsage)
1. ![BLOCKER][BLOCKER] [build.py#L255](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/8a044343e0a9d5a4497e7bd0f6fd832f8b334a59/build.py#L255): Replace print statement by built-in function. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3APrintStatementUsage)
1. ![BLOCKER][BLOCKER] [build.py#L262](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/8a044343e0a9d5a4497e7bd0f6fd832f8b334a59/build.py#L262): Replace print statement by built-in function. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3APrintStatementUsage)
1. ![BLOCKER][BLOCKER] [build.py#L263](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/8a044343e0a9d5a4497e7bd0f6fd832f8b334a59/build.py#L263): Replace print statement by built-in function. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3APrintStatementUsage)
1. ![BLOCKER][BLOCKER] [build.py#L264](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/8a044343e0a9d5a4497e7bd0f6fd832f8b334a59/build.py#L264): Replace print statement by built-in function. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3APrintStatementUsage)
1. ![BLOCKER][BLOCKER] [build.py#L265](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/8a044343e0a9d5a4497e7bd0f6fd832f8b334a59/build.py#L265): Replace print statement by built-in function. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3APrintStatementUsage)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: build.py [hunk 1]
```diff
@@ -247,6 +256,8 @@ def do_operation(self, operation) :
 				exit_code = self.do_codecov()
 			else :
 				command = self.xcodebuild_command(operation, use_xcpretty)
+				if (os.environ.get('TRAVIS') == "true" and operation == "build" and self.use_build_wrapper == "true") :
+					command = "build-wrapper-macosx-x86 --out-dir build-wrapper-output " + command
 				print command
```
Comment (partnerinflight) [inline build.py]: BLOCKER Replace print statement by built-in function. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3APrintStatementUsage)

**Comment by partnerinflight** [comment]
SonarQube analysis reported 23 issues
* ![BLOCKER][BLOCKER] 18 blocker
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] [build.py#L96](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/0ddc890cc2eb65c8fd99bb223a3268a9c8966f90/build.py#L96): Replace print statement by built-in function. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3APrintStatementUsage)
1. ![BLOCKER][BLOCKER] [build.py#L97](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/0ddc890cc2eb65c8fd99bb223a3268a9c8966f90/build.py#L97): Replace print statement by built-in function. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3APrintStatementUsage)
1. ![BLOCKER][BLOCKER] [build.py#L100](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/0ddc890cc2eb65c8fd99bb223a3268a9c8966f90/build.py#L100): Replace print statement by built-in function. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3APrintStatementUsage)
1. ![BLOCKER][BLOCKER] [build.py#L103](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/0ddc890cc2eb65c8fd99bb223a3268a9c8966f90/build.py#L103): Replace print statement by built-in function. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3APrintStatementUsage)
1. ![BLOCKER][BLOCKER] [build.py#L105](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/0ddc890cc2eb65c8fd99bb223a3268a9c8966f90/build.py#L105): Replace print statement by built-in function. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3APrintStatementUsage)
1. ![BLOCKER][BLOCKER] [build.py#L268](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/0ddc890cc2eb65c8fd99bb223a3268a9c8966f90/build.py#L268): Replace print statement by built-in function. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3APrintStatementUsage)
1. ![BLOCKER][BLOCKER] [build.py#L269](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/0ddc890cc2eb65c8fd99bb223a3268a9c8966f90/build.py#L269): Replace print statement by built-in function. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3APrintStatementUsage)
1. ![BLOCKER][BLOCKER] [build.py#L270](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/0ddc890cc2eb65c8fd99bb223a3268a9c8966f90/build.py#L270): Replace print statement by built-in function. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3APrintStatementUsage)
1. ![BLOCKER][BLOCKER] [build.py#L271](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/0ddc890cc2eb65c8fd99bb223a3268a9c8966f90/build.py#L271): Replace print statement by built-in function. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3APrintStatementUsage)
1. ![BLOCKER][BLOCKER] [build.py#L297](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/0ddc890cc2eb65c8fd99bb223a3268a9c8966f90/build.py#L297): Replace print statement by built-in function. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3APrintStatementUsage)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by partnerinflight** [comment]
SonarQube analysis reported 5 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [build.py#L182](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/0ddc890cc2eb65c8fd99bb223a3268a9c8966f90/build.py#L182): Refactor this function to reduce its Cognitive Complexity from 16 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS3776)
1. ![MAJOR][MAJOR] [build.py#L193](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/0ddc890cc2eb65c8fd99bb223a3268a9c8966f90/build.py#L193): Merge this if statement with the enclosing one. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS1066)
1. ![MAJOR][MAJOR] [build.py#L201](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/0ddc890cc2eb65c8fd99bb223a3268a9c8966f90/build.py#L201): Merge this if statement with the enclosing one. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS1066)
1. ![MINOR][MINOR] [build.py#L39](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/0ddc890cc2eb65c8fd99bb223a3268a9c8966f90/build.py#L39): Rename class "tclr" to match the regular expression ^[A-Z_][a-zA-Z0-9]+$. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS101)
1. ![MINOR][MINOR] [build.py#L182](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/0ddc890cc2eb65c8fd99bb223a3268a9c8966f90/build.py#L182): Rename this parameter "printName" to match the regular expression ^[_a-z][a-z0-9_]*$. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS117)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by partnerinflight** [comment]
SonarQube analysis reported 5 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [build.py#L177](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/ddaf94821d8eb292276b22bf23aaa1ee23fd559c/build.py#L177): Refactor this function to reduce its Cognitive Complexity from 16 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS3776)
1. ![MAJOR][MAJOR] [build.py#L188](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/ddaf94821d8eb292276b22bf23aaa1ee23fd559c/build.py#L188): Merge this if statement with the enclosing one. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS1066)
1. ![MAJOR][MAJOR] [build.py#L196](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/ddaf94821d8eb292276b22bf23aaa1ee23fd559c/build.py#L196): Merge this if statement with the enclosing one. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS1066)
1. ![MINOR][MINOR] [build.py#L39](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/ddaf94821d8eb292276b22bf23aaa1ee23fd559c/build.py#L39): Rename class "tclr" to match the regular expression ^[A-Z_][a-zA-Z0-9]+$. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS101)
1. ![MINOR][MINOR] [build.py#L177](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/ddaf94821d8eb292276b22bf23aaa1ee23fd559c/build.py#L177): Rename this parameter "printName" to match the regular expression ^[_a-z][a-z0-9_]*$. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS117)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by partnerinflight** [comment]
SonarQube analysis reported 5 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [build.py#L177](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/10240068c2f407177894b9763b7c77e86053622f/build.py#L177): Refactor this function to reduce its Cognitive Complexity from 16 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS3776)
1. ![MAJOR][MAJOR] [build.py#L188](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/10240068c2f407177894b9763b7c77e86053622f/build.py#L188): Merge this if statement with the enclosing one. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS1066)
1. ![MAJOR][MAJOR] [build.py#L196](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/10240068c2f407177894b9763b7c77e86053622f/build.py#L196): Merge this if statement with the enclosing one. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS1066)
1. ![MINOR][MINOR] [build.py#L39](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/10240068c2f407177894b9763b7c77e86053622f/build.py#L39): Rename class "tclr" to match the regular expression ^[A-Z_][a-zA-Z0-9]+$. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS101)
1. ![MINOR][MINOR] [build.py#L177](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/10240068c2f407177894b9763b7c77e86053622f/build.py#L177): Rename this parameter "printName" to match the regular expression ^[_a-z][a-z0-9_]*$. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS117)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by partnerinflight** [comment]
SonarQube analysis reported 5 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [build.py#L178](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/0a6d4124dea69b20614340d946f3dff57e0f83da/build.py#L178): Refactor this function to reduce its Cognitive Complexity from 16 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS3776)
1. ![MAJOR][MAJOR] [build.py#L189](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/0a6d4124dea69b20614340d946f3dff57e0f83da/build.py#L189): Merge this if statement with the enclosing one. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS1066)
1. ![MAJOR][MAJOR] [build.py#L197](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/0a6d4124dea69b20614340d946f3dff57e0f83da/build.py#L197): Merge this if statement with the enclosing one. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS1066)
1. ![MINOR][MINOR] [build.py#L39](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/0a6d4124dea69b20614340d946f3dff57e0f83da/build.py#L39): Rename class "tclr" to match the regular expression ^[A-Z_][a-zA-Z0-9]+$. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS101)
1. ![MINOR][MINOR] [build.py#L178](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/0a6d4124dea69b20614340d946f3dff57e0f83da/build.py#L178): Rename this parameter "printName" to match the regular expression ^[_a-z][a-z0-9_]*$. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS117)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by partnerinflight** [comment]
SonarQube analysis reported 5 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [build.py#L178](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/ebaa028c700ee6f5a1be4366f12d7f73c418fc2d/build.py#L178): Refactor this function to reduce its Cognitive Complexity from 16 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS3776)
1. ![MAJOR][MAJOR] [build.py#L189](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/ebaa028c700ee6f5a1be4366f12d7f73c418fc2d/build.py#L189): Merge this if statement with the enclosing one. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS1066)
1. ![MAJOR][MAJOR] [build.py#L197](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/ebaa028c700ee6f5a1be4366f12d7f73c418fc2d/build.py#L197): Merge this if statement with the enclosing one. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS1066)
1. ![MINOR][MINOR] [build.py#L39](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/ebaa028c700ee6f5a1be4366f12d7f73c418fc2d/build.py#L39): Rename class "tclr" to match the regular expression ^[A-Z_][a-zA-Z0-9]+$. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS101)
1. ![MINOR][MINOR] [build.py#L178](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/ebaa028c700ee6f5a1be4366f12d7f73c418fc2d/build.py#L178): Rename this parameter "printName" to match the regular expression ^[_a-z][a-z0-9_]*$. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS117)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: build.py [hunk 2]
```diff
@@ -35,7 +36,7 @@
 
 use_xcpretty = True
 
-class tclr:
+class colorValues:
```
Comment (partnerinflight) [inline build.py]: MINOR Rename class "colorValues" to match the regular expression ^[A-Z_][a-zA-Z0-9]+$. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS101)

**Comment by partnerinflight** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [build.py#L178](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/8bf671abf94dbb7ee8c643e6e3c844f8ff884eb5/build.py#L178): Rename this parameter "printName" to match the regular expression ^[_a-z][a-z0-9_]*$. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS117)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: ADAL/src/ADAuthenticationError.m [hunk 3]
```diff
@@ -236,8 +237,7 @@ + (ADAuthenticationError*)errorFromNonHttpsRedirect:(NSUUID *)correlationId
                                                  correlationId:correlationId];
 }
 
-+ (ADAuthenticationError *)keychainErrorFromOperation:(NSString *)operation
-                                               status:(OSStatus)status
++ (ADAuthenticationError *)keychainErrorFromOperation:(NSString *)operation                                               status:(OSStatus)status
```
Comment (partnerinflight) [inline ADAL/src/ADAuthenticationError.m]: MAJOR Split this 145 characters long line (which is greater than 120 authorized). [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3ALineLength)

File: ADAL/src/ADAuthenticationError.m [hunk 4]
```diff
@@ -87,7 +87,8 @@ - (id)initInternalWithDomain:(NSString *)domain
     {
         // If we're getting nil back here we have bigger problems and the logging below is going to fail anyways.`
         return nil;
-    }
+    }
```
Comment (partnerinflight) [inline ADAL/src/ADAuthenticationError.m]: MINOR Remove the useless trailing whitespaces at the end of this line. [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3AS1131)

File: ADAL/src/ADAuthenticationError.m [hunk 5]
```diff
@@ -87,7 +87,8 @@ - (id)initInternalWithDomain:(NSString *)domain
     {
         // If we're getting nil back here we have bigger problems and the logging below is going to fail anyways.`
         return nil;
-    }
+    }  
+
```
Comment (partnerinflight) [inline ADAL/src/ADAuthenticationError.m]: MINOR Remove the useless trailing whitespaces at the end of this line. [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3AS1131)

File: ADAL/src/ADAuthenticationError.m [hunk 6]
```diff
@@ -87,7 +87,8 @@ - (id)initInternalWithDomain:(NSString *)domain
...
```
Comment (partnerinflight) [inline ADAL/src/ADAuthenticationError.m]: MINOR Remove the useless trailing whitespaces at the end of this line. [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3AS1131)

**Comment by partnerinflight** [comment]
SonarQube analysis reported 22 issues
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 17 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ADAuthenticationError.m#L69](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/6da3e307682c3d8d2275dea8ef91a17b10467183/ADAL/src/ADAuthenticationError.m#L69): Split this 124 characters long line (which is greater than 120 authorized). [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3ALineLength)
1. ![MAJOR][MAJOR] [ADAuthenticationError.m#L83](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/6da3e307682c3d8d2275dea8ef91a17b10467183/ADAL/src/ADAuthenticationError.m#L83): Remove this hazardous cast. [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3AS856)
1. ![MAJOR][MAJOR] [ADAuthenticationError.m#L99](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/6da3e307682c3d8d2275dea8ef91a17b10467183/ADAL/src/ADAuthenticationError.m#L99): Split this 175 characters long line (which is greater than 120 authorized). [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3ALineLength)
1. ![MAJOR][MAJOR] [ADAuthenticationError.m#L293](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/6da3e307682c3d8d2275dea8ef91a17b10467183/ADAL/src/ADAuthenticationError.m#L293): This function has 32 returns, which is more than the 3 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3AS1142)
1. ![MINOR][MINOR] [ADAuthenticationError.m#L57](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/6da3e307682c3d8d2275dea8ef91a17b10467183/ADAL/src/ADAuthenticationError.m#L57): Remove the useless trailing whitespaces at the end of this line. [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3AS1131)
1. ![MINOR][MINOR] [ADAuthenticationError.m#L66](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/6da3e307682c3d8d2275dea8ef91a17b10467183/ADAL/src/ADAuthenticationError.m#L66): Remove the useless trailing whitespaces at the end of this line. [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3AS1131)
1. ![MINOR][MINOR] [ADAuthenticationError.m#L68](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/6da3e307682c3d8d2275dea8ef91a17b10467183/ADAL/src/ADAuthenticationError.m#L68): Remove the useless trailing whitespaces at the end of this line. [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3AS1131)
1. ![MINOR][MINOR] [ADAuthenticationError.m#L85](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/6da3e307682c3d8d2275dea8ef91a17b10467183/ADAL/src/ADAuthenticationError.m#L85): Remove the useless trailing whitespaces at the end of this line. [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3AS1131)
1. ![MINOR][MINOR] [ADAuthenticationError.m#L95](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/6da3e307682c3d8d2275dea8ef91a17b10467183/ADAL/src/ADAuthenticationError.m#L95): Remove the useless trailing whitespaces at the end of this line. [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3AS1131)
1. ![MINOR][MINOR] [ADAuthenticationError.m#L99](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/6da3e307682c3d8d2275dea8ef91a17b10467183/ADAL/src/ADAuthenticationError.m#L99): Remove this unused "message" local variable. [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3AS1481)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by partnerinflight** [comment]
SonarQube analysis reported 22 issues
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 17 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ADAuthenticationError.m#L69](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/ab5a62b705241c551ebe175d098d96acc42e5a36/ADAL/src/ADAuthenticationError.m#L69): Split this 124 characters long line (which is greater than 120 authorized). [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3ALineLength)
1. ![MAJOR][MAJOR] [ADAuthenticationError.m#L83](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/ab5a62b705241c551ebe175d098d96acc42e5a36/ADAL/src/ADAuthenticationError.m#L83): Remove this hazardous cast. [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3AS856)
1. ![MAJOR][MAJOR] [ADAuthenticationError.m#L99](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/ab5a62b705241c551ebe175d098d96acc42e5a36/ADAL/src/ADAuthenticationError.m#L99): Split this 175 characters long line (which is greater than 120 authorized). [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3ALineLength)
1. ![MAJOR][MAJOR] [ADAuthenticationError.m#L293](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/ab5a62b705241c551ebe175d098d96acc42e5a36/ADAL/src/ADAuthenticationError.m#L293): This function has 32 returns, which is more than the 3 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3AS1142)
1. ![MINOR][MINOR] [ADAuthenticationError.m#L57](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/ab5a62b705241c551ebe175d098d96acc42e5a36/ADAL/src/ADAuthenticationError.m#L57): Remove the useless trailing whitespaces at the end of this line. [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3AS1131)
1. ![MINOR][MINOR] [ADAuthenticationError.m#L66](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/ab5a62b705241c551ebe175d098d96acc42e5a36/ADAL/src/ADAuthenticationError.m#L66): Remove the useless trailing whitespaces at the end of this line. [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3AS1131)
1. ![MINOR][MINOR] [ADAuthenticationError.m#L68](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/ab5a62b705241c551ebe175d098d96acc42e5a36/ADAL/src/ADAuthenticationError.m#L68): Remove the useless trailing whitespaces at the end of this line. [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3AS1131)
1. ![MINOR][MINOR] [ADAuthenticationError.m#L85](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/ab5a62b705241c551ebe175d098d96acc42e5a36/ADAL/src/ADAuthenticationError.m#L85): Remove the useless trailing whitespaces at the end of this line. [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3AS1131)
1. ![MINOR][MINOR] [ADAuthenticationError.m#L95](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/ab5a62b705241c551ebe175d098d96acc42e5a36/ADAL/src/ADAuthenticationError.m#L95): Remove the useless trailing whitespaces at the end of this line. [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3AS1131)
1. ![MINOR][MINOR] [ADAuthenticationError.m#L99](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/ab5a62b705241c551ebe175d098d96acc42e5a36/ADAL/src/ADAuthenticationError.m#L99): Remove this unused "message" local variable. [rule](https://sonarcloud.io/coding_rules#rule_key=objc%3AS1481)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by partnerinflight** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [build.py#L194](https://github.com/AzureAD/azure-activedirectory-library-for-objc/blob/f26687afd30e42b87a112fb23ee0deec3cd81c8f/build.py#L194): Rename this parameter "printName" to match the regular expression ^[_a-z][a-z0-9_]*$. [rule](https://sonarcloud.io/coding_rules#rule_key=python%3AS117)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by partnerinflight** [comment]
Hey gang. I think this PR is ready now. I'll work on getting a different PAT for github and update once I have it.

File: build.py [hunk 7]
```diff
@@ -49,9 +50,17 @@ class tclr:
 	{
 		"name" : "iOS Framework",
 		"scheme" : "ADAL",
-		"operations" : [ "build", "test", "codecov" ],
+		"operations" : [ "build" ],
 		"min_warn_codecov" : 70.0,
 		"platform" : "iOS",
+		"use_sonarcube" : "true"
```
Comment (RPangrle) [inline build.py]: After some thought, it's occurred to me, making "use_sonarcube" only apply to the build step is probably a cleaner solution. There's some caching that we do on a per-target basis that will make the script go slower if we spit things out.

Comment (partnerinflight) [reply build.py]: ok, will make the change

**Pull request merged**